### ISSUE TITLE
libeventlog: add api function wait on event in eventlog

### ISSUE THREAD

**Issue by chu11** [flux-framework/flux-core#3201]
splitting off the library component of #3198 into a separate issue, so that PR #3200 is independent of the library addition

**Comment by chu11** [comment]
per comments in prior issue

> I think that would be far preferable, since then timeouts, callbacks, and synchronous use are all similar across the API. If we provided a way to get one eventlog entry per future fulfillment, those could be passed directly to existing parsing functions

```
     if (flux_job_event_watch_get (f, &event) < 0)
        log_err_exit ("flux_job_event_watch_get");

    if (!(o = eventlog_entry_decode (event)))
        log_err_exit ("eventlog_entry_decode");

    if (eventlog_entry_parse (o, &timestamp, &name, &context) < 0)
        log_err_exit ("eventlog_entry_parse");
```
